### PR TITLE
Fix and cleanup extension import logic

### LINKED ISSUE #1140: Development installation of AMICI fails if incompatible extension exists from previous installation
... due to e.g.:
>     AttributeError: module 'amici._amici' has no attribute 'ReturnDataPtr_posteq_cpu_timeB_get'

Could check the call stack for `setup.py`, and in that case ignore `AttributeErrors` or completely skip import of `_amici` in `__init__.py`. May also simplify the current import logic.

### PR DESCRIPTION
**Pull request by dweindl**
* Don't import AMICI extension when amici is imported from setup.py (Closes #1140)

* Cleanup import logic: Removing unused SWIG-generated files and import based on existance of remaining ones

* Avoid unnecessary imports when called from setup.py

### CODE REVIEW

File: python/amici/__init__.py [hunk 1]
```diff
@@ -84,41 +106,43 @@ def _get_commit_hash():
...
-                # `setup` or after an installation with `--no-clibs`.
+if not _imported_from_setup():
+    try:
+        # Try importing AMICI SWIG-interface with HDF5 enabled
+        from . import amici
+        from .amici import *
+        hdf5_enabled = True
+        has_clibs = True
+    except ModuleNotFoundError:
+        # import from setuptools or installation with `--no-clibs`
```
Comment (FFroehlich) [inline python/amici/__init__.py:117]: can we also detect this using frame inspection, this would reduce remaining complexity substantially.

Comment (dweindl) [reply python/amici/__init__.py:117]: The first part of the comment is now obsolete (will remove). The second part is maybe misleading, it's not during installation, it's after being installed with `--no-clibs`. We could circumvent that by writing a "configuration" file during installation to include whether we have HDF5 support or have the extension at all. That would clean up that mess quite well :)

Comment (FFroehlich) [reply python/amici/__init__.py:117]: How about `rm amici_without_hdf5.py` if hdf5 was detected during installation and `rm amici.py;mv amici_without_hdf5.py amici.py` if it was not?

Comment (dweindl) [reply python/amici/__init__.py:117]: Sounds easier. Should do the job as well.
That this removes the possibility of rebuilding the extension from the installed package later on is probably not of practical relevance...

**Pull request merged**